jobs/jobs.go

```diff
@@ -121,37 +124,37 @@ func (job *Job) Kill() {
 }
 
 // Run executes the event loop for the Job
-func (job *Job) Run() {
-	ctx, cancel := context.WithCancel(context.Background())
+func (job *Job) Run(ctx context.Context) {
```

Comment:
Just a style note: rather than having a `ctx` and `ctx2`, I like using variable names that make it clear which is a parent and which is the child. So I think elsewhere you'll see I've used `pctx` for "parent context", but if you're not wild about that name you at least get the general idea.